File: 00055_DP__Jump_Game/sol2.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {

        int idx = nums.size() - 1;

        while (idx >= 0) {
            if (idx == 0)  return true;

            int last_idx = -1;
            for (int step = 1; idx - step >= 0; step++) {
                int curr = idx - step;
                if (nums[curr] >= step) {
                    last_idx = curr;
                }
            }

            if (last_idx == -1) return false;
            idx = last_idx;
        }

        return false;
    }
};

File: 00041_Hash_Table__First_Missing_Positive/sol2.py
class Solution:

    # O(1) auxiliary space solution.
    def firstMissingPositive(self, nums: List[int]) -> int:
        
        n = len(nums)

        for i in range(n):
            if nums[i] <= 0:
                nums[i] = n + 1
        
        for i in range(n):

            val = abs(nums[i])

            if val > n:
                continue

            idx = val - 1

            # Mark minus as exist
            if nums[idx] > 0:
                nums[idx] = -1 * nums[idx]
        
        for i in range(n):
            is_exist = nums[i] < 0
            if not is_exist:
                return i + 1
        return n+1


